xbmc/threads/Condition.h

```diff
@@ -38,6 +47,18 @@ namespace XbmcThreads
       lock.count = count;
     }
 
+    template<typename Rep, typename Period>
+    inline bool wait(CCriticalSection& lock,
+                     std::chrono::duration<Rep, Period> duration,
+                     std::function<bool()> predicate)
+    {
+      int count = lock.count;
+      lock.count = 0;
```

Comment:
```c++
int count = 0;
std::swap(count, lock.count);
```

Reply:
interesting, I'm hesitant to deviate from the predefined pattern though.